File: server/php/src/Infrastructure/SDKClients/HostedTokenizationClient.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\SDKClients;

use OnlinePayments\ExampleApp\Application\DTOs\HostedTokenization\ResponseDto as ResponseDto;
use OnlinePayments\ExampleApp\Application\Interfaces\SDKClients\HostedTokenizationClientInterface;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\ExceptionMapper;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\HostedTokenizationMapper;
use OnlinePayments\Sdk\Domain\CreateHostedTokenizationRequest;
use OnlinePayments\Sdk\Merchant\MerchantClient;
use Psr\Log\LoggerInterface;

final readonly class HostedTokenizationClient implements HostedTokenizationClientInterface
{
    public function __construct(
        private MerchantClient  $merchantClient,
        private LoggerInterface $logger
    )
    {
    }

    /**
     * @throws \Throwable
     */
    public function initHostedTokenization(): ResponseDto
    {
        try {
            $this->logger->info('The hosted tokenization creation has started.');

            $request = new CreateHostedTokenizationRequest();

            $response = $this->merchantClient->hostedTokenization()->createHostedTokenization($request);

            $this->logger->info(
                "Generation of the hosted tokenization ID successfully completed -\nHostedTokenizationId: {$response->getHostedTokenizationId()}."
            );

            return HostedTokenizationMapper::mapFromSdkResponse($response);
        } catch (\Exception $ex) {
            $this->logger->error('Error occurred while creating hosted tokenization', [
                'exception' => $ex->getMessage()
            ]);

            throw ExceptionMapper::map($ex);
        }
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/SDKClients/HostedTokenizationClient.cs
using Business.DTOs.HostedTokenizations;
using Business.Interfaces.SDKClients;
using Infrastructure.Mappers;
using Infrastructure.Mappers.Exceptions;
using OnlinePayments.Sdk;
using OnlinePayments.Sdk.Domain;
using OnlinePayments.Sdk.Merchant;

namespace Infrastructure.SDKClients;

public class HostedTokenizationClient(ILogger<HostedCheckoutClient> logger, IMerchantClient merchantClient)
    : IHostedTokenizationClient
{
    public async Task<GetHostedTokenizationResponseDto> InitHostedTokenizationAsync()
    {
        try
        {
            logger.LogInformation("The generation of the hosted tokenization ID has started.");
            CreateHostedTokenizationResponse response =
                await merchantClient.HostedTokenization.CreateHostedTokenization(new CreateHostedTokenizationRequest());
            logger.LogInformation(
                "Generation of the hosted tokenization ID successfully completed - HostedTokenizationId: {}.",
                response.HostedTokenizationId);
            GetHostedTokenizationResponseDto responseDto = HostedTokenizationMapper.Map(response);

            return responseDto;
        }
        catch (ApiException ex)
        {
            throw ExceptionMapper.Map(ex);
        }
    }
}